### DISCUSSION TITLE
publish port 80 outside docker container

### DISCUSSION THREAD

**Discussion by kaushalshriyan** [wardenenv/warden#589] [Help & Support]
I am running warden on Docker version 20.10.23, build 7155243 on Ubuntu 20.04.5 LTS OS
```
warden
 _       __               __
| |     / /___ __________/ /__  ____
| | /| / / __ `/ ___/ __  / _ \/ __ \
| |/ |/ / /_/ / /  / /_/ /  __/ / / /
|__/|__/\__,_/_/   \__,_/\___/_/ /_/

Warden version 0.12.0

Usage:
  command [options] [arguments]

Options:
  -h, --help        Display this help menu

Commands:
  svc               Orchestrates global services such as traefik, portainer and dnsmasq via docker-compose
  env-init          Configure environment by adding '.env' file to the current working directory
  env               Controls an environment from any point within the root project directory
  db                Interacts with the db service on an environment (see 'warden db -h' for details)
  redis             Interacts with the redis service on an environment (see 'warden redis -h' for details)
  install           Initializes or updates warden configuration on host machine
  shell             Launches into a shell within the current project environment
  debug             Launches debug enabled shell within current project environment
  sign-certificate  Signs a wildcard certificate including all passed hostnames on the SAN list
  version           Show version information
# warden version
0.12.0
#
```

Is there a way to publish port 80 outside docker container?

 a7c9e59722f8   wardenenv/nginx:1.16                     "/bin/sh -c 'envsubs…"    About an hour ago   Up About an hour         80/tcp                                                                 fiserv245_nginx_1

For example host IP:- 192.168.0.10 and port 80 to be mapped to a7c9e59722f8 and port 80

**Comment by navarr** [comment]
Since this discussion seemed to conclude on Discord, I'm moving my answer from there to here.

The best way to route to a Warden container is by exposing Traefik.  You can do this by editing the `~/.warden/.env` file and specifying

    TRAEFIK_LISTEN=0.0.0.0

And then running `warden svc up` to restart Warden's core services.  Then Traefik will listen to port 80 connections from any machine (not just localhost).

As always, be careful with this as it will expose all Warden services to any machine connecting on Port 80, which may include Portainer, Traefik, DNSMasq, et al.
